### PR TITLE
v158: Correct processing surrounding customer's lifetime value

### LINKED ISSUE #4870: Admin, customers, calculation of lifetime value fails due to inconsistent results of float casting/currency symbol location
ZC158, php 8.1.6 (don't think it matters)

Make a couple of orders.
Admin->Customers->infobox may not show a lifetime value/cause a warning,

The calculation of this uses a casting to float:
https://github.com/zencart/zencart/blob/ac1bb27387a3a49a6d52c1334ac60c8b65927378/includes/classes/Customer.php#L313

If the currency symbol is on the left, the float renders it zero.
If the currency symbol is on the right, and decimal is a point and thousands are a comma. the float renders it correctly.
If the currency symbol is on the right, and decimal is a comma and thousands are a point. the float drops everything after the  decimal comma.

### LINKED ISSUE #4871: Admin, customers: php warning from last_order order_total being passed as string in lifetime_value clause
ZC158, php  8.1.6

If the lifetime value clause is true, the last order value is passed as a string to currencies->format
https://github.com/zencart/zencart/blob/ac1bb27387a3a49a6d52c1334ac60c8b65927378/admin/customers.php#L1061

--> PHP Warning: A non-numeric value encountered in ...zencart\includes\classes\currencies.php on line 97.
https://github.com/zencart/zencart/blob/ac1bb27387a3a49a6d52c1334ac60c8b65927378/includes/classes/currencies.php#L97

### PR DESCRIPTION
**Pull request by lat9**
Fixes #4871
Fixes #4870

### CODE REVIEW

File: admin/customers.php [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ function check_form() {
 
                   if (!empty($cInfo->lifetime_value)) {
                     $contents[] = array('text' => TEXT_INFO_LIFETIME_VALUE . ' ' . $currencies->format($cInfo->lifetime_value));
-                    $contents[] = array('text' => TEXT_INFO_LAST_ORDER . ' ' . zen_date_short($cInfo->last_order['date_purchased']) . '<br>' . TEXT_INFO_ORDERS_TOTAL . ' ' . $currencies->format($cInfo->last_order['order_total'], true, $cInfo->last_order['currency'], $cInfo->last_order['currency_value']));
+                    $contents[] = array('text' => TEXT_INFO_LAST_ORDER . ' ' . zen_date_short($cInfo->last_order['date_purchased']) . '<br>' . TEXT_INFO_ORDERS_TOTAL . ' ' . (float)$currencies->format($cInfo->last_order['order_total'], true, $cInfo->last_order['currency'], (float)$cInfo->last_order['currency_value']));
```
Comment (torvista) [inline admin/customers.php:1060]: It appears the first float added is in the wrong place.

Comment (lat9) [reply admin/customers.php:1060]: Yes it was; but as noted for #4870, that line in `customers.php` isn't going to fly anyway.  I'll resubmit to correct both #4870 and #4871 in one-swell-foop.

**Comment by scottcwilson** [comment]
LMK when you feel this is ready to merge @lat9.

**Comment by lat9** [comment]
> LMK when you feel this is ready to merge @lat9.

Waiting for @torvista, since he was the original reporter of the issues.

**Comment by torvista** [comment]
This works in my tests.

**Pull request merged**